### PR TITLE
Update GenAI SIG meeting time, GC sponsor and other cleanup

### PR DESCRIPTION
**Pull request by lmolkova**
Follow up on https://github.com/open-telemetry/community/pull/2326

- update meeting time Wed 10am PST -> Thu 10am PST
- update GC sponsor and liaison (@tedsuo is the sponsor now)
- rename LLM to GenAI and reflect new approver group name and slack channel

### CODE REVIEW

**Review by nirga — APPROVED** [review]

File: docs/using-github-extensions.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ Extensions, apps, or bots that can **arbitrarily modify code are NOT allowed**.
 
 ## Writing your GitHub Action workflows
 
-Many GitHub Action workflows do not require a dedicated GitHub account. Good examples are auto-assign workflows for issues and PRs such as [the one used in semantic convention repo](https://github.com/open-telemetry/semantic-conventions/blob/main/.github/workflows/auto-assign-issue.yml).
+Many GitHub Action workflows do not require a dedicated GitHub account. Good examples are auto-assign workflows for PRs such as [the one used in specification repo](https://github.com/open-telemetry/opentelemetry-specification/blob/main/.github/auto_assign_pr.yml).
```
Comment (lmolkova) [inline docs/using-github-extensions.md:21]: unrelated to this PR,  fixes CI failure - the workflow was recently removed in https://github.com/open-telemetry/semantic-conventions/pull/1497

File: projects/gen-ai.md [hunk 2]
```diff
@@ -123,9 +123,9 @@ Other languages are not in the scope (yet).
 
 ## Meeting Times
 
-We'll use GenAI LLM Semantic Convention meetings until more suitable time for new participants is determined:
+We'll use GenAI Semantic Convention meetings until more suitable time for new participants is determined:
 
-- **Every week**: Wednesdays at 10 AM PST - Looking for alternative time to accommodate more time zones - https://strawpoll.com/e6Z2ApEKXgN
+- **Every week**: Thursday at 10 AM PST
 - **Every other week**: Thursday 14:00 UTC+8
```
Comment (trask) [inline projects/gen-ai.md:129]:
```suggestion
- **Every other week**: Thursday at 10 AM PST
- **Every other week**: Thursday 14:00 UTC+8
```

**Review by trask — APPROVED** [review]

File: projects/gen-ai.md [hunk 3]
```diff
@@ -123,9 +123,9 @@ Other languages are not in the scope (yet).
 
 ## Meeting Times
 
-We'll use GenAI LLM Semantic Convention meetings until more suitable time for new participants is determined:
+We'll use GenAI Semantic Convention meetings until more suitable time for new participants is determined:
 
-- **Every week**: Wednesdays at 10 AM PST - Looking for alternative time to accommodate more time zones - https://strawpoll.com/e6Z2ApEKXgN
+- **Every other week**: Thursday at 10 AM PST
 - **Every other week**: Thursday 14:00 UTC+8
```
Comment (drewby) [inline projects/gen-ai.md:129]: If there is no push back to the proposal, can we update this to Thursday at 16:00 PST.

Comment (lmolkova) [reply projects/gen-ai.md:129]: yep, let's bring it up again tomorrow and give folks a bit of time to respond :)

**Review by karthikscale3 — APPROVED** [review]

File: projects/gen-ai.md (hunk 3 above)
Comment (lmolkova) [reply projects/gen-ai.md:129]: let's do it as a follow up

**Pull request merged**